### PR TITLE
Allow for event deletion

### PR DESCRIPTION
**Pull request by agentd00nut**
Multiple calendars can be added to the sourceCalendars global.

Events found in a users calendar not found in a feed during a sync can be removed, controlled by global flag.

Events no longer store feed ids in the description and use a custom "FID" metadata tag.

Event matching is based on this "FID" tag.

### CODE REVIEW

File: ICS-ICAL Sync/Code.gs [hunk 1]
```diff
@@ -20,13 +20,14 @@ var sourceCalendars={
...
 var addEventsToCalendar = true; //If you turn this to "false", you can check the log (View > Logs) to make sure your events are being read correctly before turning this on
+var modifyExistingEvents = true; // If you turn this to "false", any event in the feed that was modified after being added to the calendar will not update
 var removeEventsFromCalendar = true; //If you turn this to "true", any event in the calendar not found in the feed will be removed.  
 var createMissingCalendar=true;  // If the calendar name is not found we create it instead of throwing an error, only if the sourceUrl works.
 var addAlerts = true; //Whether to add the ics/ical alerts as notifications on the Google Calendar events
 var descriptionAsTitles = false; //Whether to use the ics/ical descriptions as titles (true) or to use the normal titles as titles (false)
-var defaultDuration = 60; //Default duration (in minutes) in case the event is missing an end specification in the ICS/ICAL file
+var defaultDuration = 30; //Default duration (in minutes) in case the event is missing an end specification in the ICS/ICAL file
```
Comment (derekantrican) [inline ICS-ICAL Sync/Code.gs]: What is the argument for making this change? I don't see a reason to change the default value of `defaultDuration`

File: ICS-ICAL Sync/Code.gs [hunk 2]
```diff
@@ -4,20 +4,34 @@
...
 // --------------- SETTINGS ---------------
-var sourceCalendarURL = ""; //The ics/ical url that you want to get events from
-var targetCalendarName = ""; //The name of the Google Calendar you want to add events to
-var howFrequent = 15; //What interval (minutes) to run this script on to check for new events
+
+// var sourceCalendars = { targetCalendarName:"sourceCalendarURL"}
+//     "targetCalendarName" is The name of the Google Calendar you want to add events to
+//     "sourceCalendarURL" is The ics/ical url that you want to get events from
+
+var sourceCalendars={
```
Comment (derekantrican) [inline ICS-ICAL Sync/Code.gs]: Let's do this change (support for multiple calendars) in a separate pull request. Log an issue for it

File: ICS-ICAL Sync/Code.gs [hunk 3]
```diff
@@ -4,20 +4,34 @@
...
+};
+
+
+// Currently global settings are applied to all sourceCalendars.  
+
+var howFrequent = 20; //What interval (minutes) to run this script on to check for new events
 var addEventsToCalendar = true; //If you turn this to "false", you can check the log (View > Logs) to make sure your events are being read correctly before turning this on
+var modifyExistingEvents = true; // If you turn this to "false", any event in the feed that was modified after being added to the calendar will not update
+var removeEventsFromCalendar = true; //If you turn this to "true", any event in the calendar not found in the feed will be removed.  
+var createMissingCalendar=true;  // If the calendar name is not found we create it instead of throwing an error, only if the sourceUrl works.
```
Comment (derekantrican) [inline ICS-ICAL Sync/Code.gs]: I think we should just do this anyway. Don't have it be a user setting, just create the calendar if doesn't exist

File: ICS-ICAL Sync/Code.gs [hunk 4]
```diff
@@ -108,33 +146,49 @@ function main() {
...
-    Logger.log("Description: " + events[i].description);
-    Logger.log("Start: " + events[i].startTime);
-    Logger.log("End: " + events[i].endTime);
-    
-    for (var j = 0; j < events[i].reminderTimes.length; j++)
-      Logger.log("Reminder: " + events[i].reminderTimes[j] + " seconds before");
-    
-    Logger.log("");
-  }
+//  for (var i = 0; i < events.length; i++){
```
Comment (derekantrican) [inline ICS-ICAL Sync/Code.gs]: Why were these lines (159-170) commented out? A change made during your testing?

File: ICS-ICAL Sync/Code.gs [hunk 5]
```diff
@@ -32,28 +46,50 @@ var email = ""; //OPTIONAL: If "emailWhenAdded" is set to true, you will need to
...
+
 //---------------- DO NOT EDIT BELOW HERE UNLESS YOU REALLY KNOW WHAT YOU'RE DOING --------------------
 function Install(){
   ScriptApp.newTrigger("main").timeBased().everyMinutes(howFrequent).create();
 }
 
-function main() {  
+function main(){
+  
+  for( var calendar in sourceCalendars){
```
Comment (derekantrican) [inline ICS-ICAL Sync/Code.gs]: Again, let's do multi-calendar support in a different pull request

File: ICS-ICAL Sync/Code.gs [hunk 6]
```diff
@@ -32,28 +46,50 @@ var email = ""; //OPTIONAL: If "emailWhenAdded" is set to true, you will need to
...
+  
+  
   //------------------------ Error checking ------------------------
   if(response[0] == "That calendar does not exist.")
     throw "[ERROR] Incorrect ics/ical URL";
   
-  if(targetCalendar == null)
-    throw "[ERROR] Could not find calendar of name \"" + targetCalendarName + "\"";
+  if(targetCalendar == null){
+    if(createMissingCalendar){
```
Comment (derekantrican) [inline ICS-ICAL Sync/Code.gs]: Like above comment - just do this. Don't need a user setting for it

File: ICS-ICAL Sync/Code.gs (hunk 1 above)
Comment (agentd00nut) [reply ICS-ICAL Sync/Code.gs]: No good reason.  Just forgot i changed it.

File: ICS-ICAL Sync/Code.gs (hunk 4 above)
Comment (agentd00nut) [reply ICS-ICAL Sync/Code.gs]: Sorry yea, i kept having to scroll past all the output when debugging and forgot to flip it back.

**Comment by derekantrican** [comment]
@agentd00nut I want to revive this again and get it pulled through. I'll have some time this afternoon (around 4pm-6pm PST). Here's a Slack channel where we can discuss this better: https://join.slack.com/t/gasicssync/shared_invite/enQtMzc2MTk4MTkzOTEwLTViNGU4NjhkZmRhMjliMzY2NjY2ZWJjNDdkZGY3NTBiMTlmMTViMDZiZTE0NGVkMjFmMTU3NzkzZWQ2ZGY3NjM

**Comment by derekantrican** [comment]
Resolved #5

**Pull request merged**